ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "../PrjFSKext/kernel-header-wrappers/vnode.h"
+#include "../PrjFSKext/kernel-header-wrappers/mount.h"
+#include "../PrjFSKext/public/FsidInode.h"
+#include <memory>
+#include <string>
+
+typedef std::shared_ptr<mount> MountPointer;
+typedef std::shared_ptr<vnode> VnodePointer;
+typedef std::weak_ptr<vnode> VnodeWeakPointer;
```

Comment:
Style suggestion: This might just be me, but I'm not sure we're gaining much readability by typedefing  `shared_ptr` and `weak_ptr`.

If we are going to keep the typedefs then I think we should include "Shared" in the names of the `shared_ptr` types (as we have "Weak" in the `weak_ptr` type).